### PR TITLE
Remove `:query` and `:entity` fiddles types

### PR DESCRIPTION
**Pull request by ggeoffrey**
Also remove:
- `fiddle/type`
- `fiddle/query`
- `fiddle/pull`
- `fiddle/pull-database`
- `QueryRequest`, `EvalRequest`, `EntityRequest`
- `hydrate-request*`
- `request-for-fiddle+`

### CODE REVIEW

File: src/hyperfiddle/ide/authenticate.cljc [hunk 1]
```diff
@@ -48,10 +47,8 @@
                                  (-> config :auth0 :client-secret)
                                  (-> config :auth0 :domain))]
                     (p/do! (verify encoded-id-token)))
-         basis (io/sync io #{"$users"})
-         user-record (->> (map->EntityRequest {:e [:user/sub (:sub id-token)]
-                                               :db ["$users" hf/root-pid]
-                                               :pull-exp [:db/id :user/user-id :user/created-date]})
+         basis (io/sync io #{"$" "$users"})
+         user-record (->> `((subject ~(:sub id-token)) ~hf/root-pid)
```
Comment (dustingetz) [inline src/hyperfiddle/ide/authenticate.cljc:51]: So hydrate-one! takes a list of sexp, nice

File: src/hyperfiddle/io/datomic/hydrate_requests.clj [hunk 2]
```diff
@@ -209,27 +134,52 @@
...
+;;;;;;;;;;;;;;;
+
+(defn max-arity [avar]
+  (or (some->> avar meta :arglists
+               (map (fn [args] (count (remove #{'&} args))))
+               (apply max))
+      0))
+
+(defmethod hf/defaults :default [[f & args]]
+  (cons f (data/zipseq args (repeat (max-arity (find-var f)) nil))))
```
Comment (dustingetz) [inline src/hyperfiddle/io/datomic/hydrate_requests.clj:149]: I see you are nil padding to max arity, but why?

File: src/hyperfiddle/io/datomic/hydrate_requests.clj [hunk 3]
```diff
@@ -209,27 +134,52 @@
...
+
+(defmethod hf/view-defaults :default [route] route)
+
+(defn eval-as-fexpr!
+  "Like `clojure.core/eval`, but don't evaluate operands.
+  https://en.wikipedia.org/wiki/Fexpr ."
+  [[fn-sym & args]]
+  (-> (find-var fn-sym)
+      (deref)
+      (apply args)))
```
Comment (dustingetz) [inline src/hyperfiddle/io/datomic/hydrate_requests.clj:159]: Do you still believe this is an fexpr?

File: src/hyperfiddle/schema.cljc [hunk 4]
```diff
@@ -1,37 +1,39 @@
...
-                                                              {:db/valueType [:db/ident]
-                                                               :db/cardinality [:db/ident]
-                                                               :db/unique [:db/ident]}])
-                                           :where [:db.part/db :db.install/attribute ?attr]]
-                                         [(runtime/db rt pid dbname)]
-                                         {:limit -1}))
+  (let [dbnames  (-> (hf/databases (hf/domain rt))
+                     keys
+                     vec)
+        requests (mapv (fn [dbname] `((query-schema ~dbname) ~pid))
```
Comment (dustingetz) [inline src/hyperfiddle/schema.cljc:18]: well that's lovely

File: src/hyperfiddle/schema.cljc [hunk 5]
```diff
@@ -1,37 +1,39 @@
...
                        dbnames)]
     (-> (io/hydrate-requests (hf/io rt) local-basis partitions requests)
         (p/then (fn [{:keys [pulled-trees]}]
                   (->> pulled-trees                         ; add local schemas from spec here?
                        (map (fn [pulled-tree+]
-                              (-> (cats/fmap #(contrib.datomic/indexed-schema (mapv first %)) pulled-tree+)
+                              (-> (cats/fmap #(->> % (last) (mapv first) (contrib.datomic/indexed-schema)) pulled-tree+)
                                   (either/branch exception/failure exception/success))))
```
Comment (dustingetz) [inline src/hyperfiddle/schema.cljc:25]: we should remove cats/Either as we can

File: src/hyperfiddle/ui.cljs [hunk 6]
```diff
@@ -237,21 +237,7 @@ User renderers should not be exposed to the reaction."
...
-                          ; just check if a request can be built
-                          (base/request-for-fiddle+ (:runtime ctx) (:partition-id ctx) ?route ?fiddle)
-                          (cats/fmap (constantly ?route)))
-                 :entity (let [[$1 :as params] (::route/datomic-args ?route)]
-                           ; todo just run base/request-for-fiddle+
-                           (if (not= nil $1)                ; todo check conn
-                             (either/right ?route)
-                             (either/left {:message "malformed entity param" :data {:params params}})))
-                 (either/right ?route)))))
+      (>>= (fn [?route] (either/right ?route))) ;; FIXME
```
Comment (dustingetz) [inline src/hyperfiddle/ui.cljs:240]: what's this

File: src/hyperfiddle/io/datomic/hydrate_requests.clj (hunk 2 above)
Comment (ggeoffrey) [reply src/hyperfiddle/io/datomic/hydrate_requests.clj:149]: We agreed fiddles should expose a single arity. Providing multiple arities is a mistake, will issue a warning, and only the longest one will be considered by HF. So nil padding to the max arity is the default behavior.
Of course if you only provide a single arity, then it's also the max arity.

File: src/hyperfiddle/io/datomic/hydrate_requests.clj (hunk 3 above)
Comment (ggeoffrey) [reply src/hyperfiddle/io/datomic/hydrate_requests.clj:159]: No, not anymore. But I'm still searching for a better name.

File: src/hyperfiddle/schema.cljc (hunk 4 above)
Comment (ggeoffrey) [reply src/hyperfiddle/schema.cljc:18]: I was worried the mental overhead of quoting and unquoting would be too high.
A classic `[route pid]` pair or `{:route route, :pid pid}` would do the job too.
What do you think?

File: src/hyperfiddle/ui.cljs (hunk 6 above)
Comment (ggeoffrey) [reply src/hyperfiddle/ui.cljs:240]:
```diff
- (>>= (fn [?route]
-              ; todo should still allow routing even if these checks fail, all that matters if it can be encoded to an url
-              ; We specifically hydrate this deep just so we can validate anchors like this.
-              (let [?fiddle (:link/fiddle @link-ref)]
-                (case (:fiddle/type ?fiddle)
-                  :query (->>
-                           ; just check if a request can be built
-                           (base/request-for-fiddle+ (:runtime ctx) (:partition-id ctx) ?route ?fiddle)
-                           (cats/fmap (constantly ?route)))
-                  :entity (let [[$1 :as params] (::route/datomic-args ?route)]
-                            ; todo just run base/request-for-fiddle+
-                            (if (not= nil $1)                ; todo check conn
-                              (either/right ?route)
-                              (either/left {:message "malformed entity param" :data {:params params}})))
-                  (either/right ?route)))))
+ (>>= (fn [?route] (either/right ?route)))
```

We need to drop the bind call entirely.

**Pull request merged**